llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -444,15 +444,36 @@ class SampleProfileMatcher {
   // the profile.
   StringMap<LocToLocMap> FuncMappings;
 
-  // Profile mismatching statstics.
+  // Match state for an anchor/callsite.
+  enum class MatchState {
+    Matched = 0,
+    Mismatched = 1,
+    // Stay Matched after profile matching.
+    StayMatched = 2,
```

Comment:
Sure, I lean towards the second one(state change), thinking it can align with the name of the stats counter such as `NumRecoveredCallsites`, `RecoveredCallsiteSamples` 

Updated in the new version, thanks for the suggestion!